src/test/java/org/skife/jdbi/v2/sqlobject/TestClasspathStatementLocator.java

```diff
@@ -48,15 +49,32 @@ public void tearDown() throws Exception {
 
     @Test
     public void testBam() throws Exception {
-        handle.execute("insert into something (id, name) values (6, 'Martin')");
+        Something s = handle.attach(Cromulence.class).findById(6L);
+        assertThat(s.getName(), equalTo("Martin"));
+    }
+
+    @Test
+    public void testOverride() throws Exception {
+        Something s = handle.attach(SubCromulence.class).findById(6L);
+        assertThat(s.getName(), equalTo("overridden"));
+    }
 
+    @Test
+    public void testCachedOverride() throws Exception {
         Something s = handle.attach(Cromulence.class).findById(6L);
         assertThat(s.getName(), equalTo("Martin"));
+
+        // and now make sure we don't accidentally cache the statement from above
+        s = handle.attach(SubCromulence.class).findById(6L);
+        assertThat(s.getName(), equalTo("overridden"));
     }
 
     @RegisterMapper(SomethingMapper.class)
     static interface Cromulence {
         @SqlQuery
         public Something findById(@Bind("id") Long id);
     }
+
+    @RegisterMapper(SomethingMapper.class)
```

Comment:
Huh, I guess we don't look at supertypes for annotations.. we should probably fix that in v3.